### ISSUE TITLE
CA2012 wrong warning message when assign ValueTask to discard

### ISSUE THREAD

**Issue by maxkatz6** [dotnet/roslyn-analyzers#3573]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v3.0.0 (Latest)

#### Diagnostic ID

[CA2012](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2012) (link is broken due to https://github.com/dotnet/roslyn-analyzers/issues/3549)

#### Repro steps

```csharp
_ = MyMethod();

ValueTask MyMethod() => default;
```

#### Expected behavior
Warning	CA2012	ValueTask instances returned from method calls should always be used, typically awaited. Not doing so often represents a functional bug, but even if it doesn't, it can result in degraded performance if the target method pools objects for use with ValueTasks.

#### Actual behavior
Warning	CA2012	ValueTask instances returned from method calls should be directly awaited, returned, or passed as an argument to another method call. Other usage, such as storing an instance into a local or a field, is likely an indication of a bug, as ValueTask instances must only ever be consumed once.

**Comment by mavasani** [comment]
Tag @stephentoub

**Comment by maxkatz6** [comment]
Well, without discard I see now another analyzer message:
CA2012	ValueTask instances returned from method calls should always be used, typically awaited. Not doing so often represents a functional bug, but even if it doesn't, it can result in degraded performance if the target method pools objects for use with ValueTasks.

So, I shouldn't use ValueTask with discard either.
Then probably "_ = MyMethod" should produce more accurate warning?
I will update issue body and title.

**Comment by stephentoub** [comment]
I'm trying to understand the issue.  The concern isn't that we're outputting a warning here, but rather that the message we're outputting is the wrong one?

**Comment by maxkatz6** [comment]
@stephentoub 
Yes, the message is wrong.
In updated issue you can see actual message, which says that we shouldn't store ValueTask in any field. But that's wrong, because we doesn't store ValueTask in any field with discard. "Expected behavior" message will be more correct.

**Comment by stephentoub** [comment]
Thanks for reporting the issue.  I've fixed it to use a more specific error message; it was previously just using the generic one.

**Comment by maxkatz6** [comment]
@stephentoub thanks!
